### DISCUSSION TITLE
Feat: Introducing "Content Admin" Role

### DISCUSSION THREAD

**Discussion by MarcRevo** [open-webui/open-webui#5120] [Ideas]
Hello everyone,

I'm planning to submit a pull request to introduce a new user role, "**content admin**". This role is designed to enhance document management capabilities and will include specific permissions tailored to this function.

PR: https://github.com/open-webui/open-webui/pull/5121

### Proposed Changes

**1. New User Role: "Content Admin"**
   - **Description**: The "content admin" role will have permissions to manage documents and access related features.
   - **Frontend**:
      - Updates to several components to support role-based access, including `AddUserModal.svelte`, `General.svelte`, `Sidebar.svelte`, and others.
   - **Backend**:
      - Updates to role handling logic in `auths.py`.
      - Adjustments to chat access permissions and document management endpoints.
      - New utility functions to support the "content admin" role.
      
**2. Breaking Changes**
   - The introduction of this new role will necessitate updates across both backend and frontend components. Please ensure that all relevant parts of the system are reviewed and updated accordingly.
   
**3. Potential Impact**
  - The changes will impact role-based views and access controls throughout the application. I believe these updates will enhance our system's capabilities and provide better management features for document-related tasks.

### Action Required
Before I submit the pull request, I would appreciate feedback on the following:
  - **Role Definition**: Are the permissions and functionalities for the "content admin" role well-defined and suitable?
  - **Frontend Integration**: Are there any additional components or areas that should be considered for integrating the new role
  - **Backend Adjustments**: Are there other backend considerations or edge cases that should be addressed?

Please let me know your thoughts, suggestions, or concerns about these proposed changes. I am open to any feedback and eager to ensure that these updates align with the project's goals.

Thank you!

Marc

**Comment by justinh-rahb** [comment]
I might hold off on this as it might fall under the umbrella of existing Teams FR, which could result in duplicated or conflicting work.
- #2924 

Thoughts @tjbck ?

**Reply by tjbck** [reply]
Duplicate work indeed, teams feature will be addressed asap to cover this use case as well.
